### DISCUSSION TITLE
Does mantine support server components?

### DISCUSSION THREAD

**Discussion by rawand-faraidun** [mantinedev/mantine#5414] [Q&A] [answered]
in the [next-app-template](https://github.com/mantinedev/next-app-template) the `theme.ts` uses client which made me wonder if Mantine supports server components, for my project SEO is so important that is why I ask

**Comment by rtivital** [comment, marked as answer]
**I've added `'use client';` at the top of the file. Does it mean that my component won't be rendered on the server?**
No, although the naming is very confusing and unintuitive, your component will be rendered on the server (both SSR and SSG).

**What is the difference between "client" and "server" components then?**
When you set `'use client';` at the top of the file, the component becomes "client". "client" components are rendered **both on client and server**. It mean that component html will be generated on on server during SSR/SSG and then component will be hydrated on client. "server" components are rendered **only on server**, "server" component is not hydrated on client, it does not have associated representation in your JS bundle.

**Can I use Mantine components as "server" components?**
No, Mantine components cannot be used as "server" components, only as "client" component. It means that Mantine components are rendered **both on client and server**, **not only on client**.

**But why I cannot use Mantine components as "server" components?**
All Mantine components use React context for default props and Styles API features. It is not planned to remove these features. With current implementation "server" components implementation it will not be possible to use Mantine components as "server" components – all of components have `'use client';` at the top of the files. Remember that **"client" components are rendered both on client and server, not only on client**.

**Reply by rawand-faraidun** [reply]
Amazing response. thank you for clarifying, i though client ones are only rendered on client side, but with now i am okay to use Mantine knowing it won't affect SEO

**Reply by yishayweb** [reply]
> All Mantine components use React context for default props and Styles API features. It is not planned to remove these features.

@rtivital I'm not sure if this answer was before/after the migration to native css. Is there any plan now to somehow have a mode in the library (in case I use css modules) that does support Server Components in a way that it would be possible to render (at least some of library components) only on the server?
Thank you for all the amazing work you are doing :pray:

**Reply by rtivital** [reply]
https://help.mantine.dev/q/server-components
